app/models/post.rb

```diff
@@ -1,6 +1,16 @@
 # Base class for models in the application.
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
   has_many :comments
   has_many :likes
+
+  # A method that updates the posts counter for a user.
+  def update_counter_cache
+    author.update(postsCounter: author.posts.count)
+  end
+
```

Comment:
- [ ] I think it would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to `database`, then make the `update_post_counter` method private. I noticed all your updates method has the same name, it would be nice to name them differently to avoid confusion in the code 🙃 🙃 
The code can be similar to 👇 

```ruby
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```